__tests__/Unit/Components/Tasks/TaskDependency.test.tsx

```diff
@@ -0,0 +1,48 @@
+import { fireEvent, render } from '@testing-library/react';
+import TaskDependency from '@/components/taskDetails/taskDependency';
+import { taskDetailsHandler } from '../../../../__mocks__/handlers/task-details.handler';
+import { setupServer } from 'msw/node';
+import { Provider } from 'react-redux';
+import { store } from '@/app/store';
+import { TaskDependencyIds } from '../../../../__mocks__/db/tasks';
+
+const mockNavigateToTask = jest.fn();
+jest.mock('next/router', () => ({
+    useRouter: () => ({
+        push: mockNavigateToTask,
+    }),
+}));
+describe('TaskDependency', () => {
+    const server = setupServer(...taskDetailsHandler);
+    beforeAll(() => server.listen());
+    afterEach(() => server.resetHandlers());
+    afterAll(() => server.close());
+    it('should update editedDependencies state and call handleChange when dependencies change', () => {
+        const handleChange = jest.fn();
+
+        const { getByRole } = render(
+            <Provider store={store()}>
+                <TaskDependency
+                    isEditing={true}
+                    updatedDependencies={[]}
+                    handleChange={handleChange}
+                    taskDependencyIds={TaskDependencyIds}
+                />
+            </Provider>
+        );
+
+        const textarea = getByRole('textbox');
+        const event = { target: { value: 'task1, task2, task3' } };
```

Comment:
here the tasks are spaced, but below there are no spaces. Is this how the logic is supposed to be?

Reply:
Oh, I saw `trim()` - resolving